src/t9p.c

```diff
@@ -3492,12 +3498,21 @@ tr_send_now(struct t9p_context* c, struct trans_node* n)
       ERROR(c, "send: %s\n", strerror(-l));
       if (l == -EPIPE)
         t9p__conn_broken(c);
-      return -1;
+      err = -1;
+      goto error;
     }
     atomic_add_u32(&c->stats.send_cnt, 1);
     atomic_add_u32(&c->stats.total_bytes_send, n->tr.size);
   }
+
+  if (n->tr.hdata)
+    c->trans.batch(c->conn, 0);
   return 0;
+
+error:
+  if (n->tr.hdata)
+    c->trans.batch(c->conn, 0);
+  return err;
```

Comment:
nit: You could initialize `err` to 0 and merge this block

Reply:
done, thx for the suggestion!